server/ai_mediaserver.go

```diff
@@ -621,14 +622,144 @@ func (ls *LivepeerServer) StartLiveVideo() http.Handler {
+			clog.Errorf(ctx, "Error starting processing: %s", err)
+			params.liveParams.stopPipeline(err)
+			perOrchCancel()
+			return
+		}
+
+		<-params.liveParams.processing
+		perOrchCancel()
+
+		if !orchSwapper.shouldSwap(ctx) {
+			break
+		}
+		clog.Infof(ctx, "Retrying stream with a different orchestrator")
+	}
+	params.liveParams.stopPipeline(fmt.Errorf("Done processing"))
+	params.liveParams.outputWriter.Close()
+}
+
+func newParams(params *liveRequestParams) *liveRequestParams {
+	return &liveRequestParams{
+		segmentReader:          params.segmentReader,
+		outputRTMPURL:          params.outputRTMPURL,
+		mediaMTXOutputRTMPURL:  params.mediaMTXOutputRTMPURL,
+		stream:                 params.stream,
+		paymentProcessInterval: params.paymentProcessInterval,
+		requestID:              params.requestID,
+		streamID:               params.streamID,
+		pipelineID:             params.pipelineID,
+		modelID:                params.modelID,
+		stopPipeline:           params.stopPipeline,
+		sendErrorEvent:         params.sendErrorEvent,
+		processing:             make(chan struct{}),
+		outputWriter:           params.outputWriter,
+		startTime:              time.Now(),
+	}
+}
+
+func startOutput(ctx context.Context, params aiRequestParams) *multiWriter {
+	var err error
+	rMediaMTX, wMediaMTX, err := os.Pipe()
+	if err != nil {
+		params.liveParams.stop()
+		return nil
+	}
+	r, w, err := os.Pipe()
+	if err != nil {
+		params.liveParams.stop()
+		return nil
+	}
+	mWriter := &multiWriter{ctx: ctx, writers: []io.Writer{w, wMediaMTX}}
+	// Studio Output ffmpeg process
+	if params.liveParams.outputRTMPURL != "" {
+		go ffmpegOutput(ctx, params.liveParams.outputRTMPURL, r, params)
+	}
+	go ffmpegOutput(ctx, params.liveParams.mediaMTXOutputRTMPURL, rMediaMTX, params)
+	return mWriter
+}
+
+func startProcessing(ctx context.Context, params aiRequestParams, res interface{}) error {
+	resp := res.(*worker.GenLiveVideoToVideoResponse)
+
+	host := params.liveParams.sess.Transcoder()
+	pub, err := common.AppendHostname(resp.JSON200.PublishUrl, host)
+	if err != nil {
+		return fmt.Errorf("invalid publish URL: %w", err)
+	}
+	sub, err := common.AppendHostname(resp.JSON200.SubscribeUrl, host)
+	if err != nil {
+		return fmt.Errorf("invalid subscribe URL: %w", err)
+	}
+	control, err := common.AppendHostname(*resp.JSON200.ControlUrl, host)
+	if err != nil {
+		return fmt.Errorf("invalid control URL: %w", err)
+	}
+	events, err := common.AppendHostname(*resp.JSON200.EventsUrl, host)
+	if err != nil {
+		return fmt.Errorf("invalid events URL: %w", err)
+	}
+	clog.V(common.VERBOSE).Infof(ctx, "pub %s sub %s control %s events %s", pub, sub, control, events)
+
+	onFirstSegment := func() {
+		delayMs := time.Since(params.liveParams.startTime).Milliseconds()
+		if monitor.Enabled {
+			monitor.AIFirstSegmentDelay(delayMs, params.liveParams.sess.OrchestratorInfo)
+			monitor.SendQueueEventAsync("stream_trace", map[string]interface{}{
```

Comment:
We probably need to make sure this event change doesn't break the data pipeline. I guess we might now send this multiple times for the same stream, and depending on how we were measuring it before the observations might be shorter? (in case we changed the time where se set the `startTime`)